settings/memcache.settings.php

```diff
@@ -12,78 +12,82 @@
-          'arguments' => ['@memcache.config'],
-        ],
-        'memcache.backend.cache.container' => [
-          'class' => 'Drupal\memcache\DrupalMemcacheFactory',
-          'factory' => ['@memcache.backend.cache.factory', 'get'],
-          'arguments' => ['container'],
-        ],
-        'lock.container' => [
-          'class' => 'Drupal\memcache\Lock\MemcacheLockBackend',
-          'arguments' => ['container', '@memcache.backend.cache.container'],
-        ],
-        'cache_tags_provider.container' => [
-          'class' => 'Drupal\Core\Cache\DatabaseCacheTagsChecksum',
-          'arguments' => ['@database'],
-        ],
-        'cache.container' => [
-          'class' => 'Drupal\memcache\MemcacheBackend',
-          'arguments' => [
-            'container',
-            '@memcache.backend.cache.container',
-            '@lock.container',
-            '@memcache.config',
-            '@cache_tags_provider.container',
-          ],
+  // Bootstrap cache.container with memcache rather than database.
+  $settings['bootstrap_container_definition'] = [
+    'parameters' => [],
+    'services' => [
+      'database' => [
+        'class' => 'Drupal\Core\Database\Connection',
+        'factory' => 'Drupal\Core\Database\Database::getConnection',
+        'arguments' => ['default'],
+      ],
+      'settings' => [
+        'class' => 'Drupal\Core\Site\Settings',
+        'factory' => 'Drupal\Core\Site\Settings::getInstance',
+      ],
+      'memcache.config' => [
+        'class' => 'Drupal\memcache\DrupalMemcacheConfig',
+        'arguments' => ['@settings'],
+      ],
+      'memcache.backend.cache.factory' => [
+        'class' => 'Drupal\memcache\DrupalMemcacheFactory',
+        'arguments' => ['@memcache.config'],
+      ],
+      'memcache.backend.cache.container' => [
+        'class' => 'Drupal\memcache\DrupalMemcacheFactory',
+        'factory' => ['@memcache.backend.cache.factory', 'get'],
+        'arguments' => ['container'],
+      ],
+      'lock.container' => [
+        'class' => 'Drupal\memcache\Lock\MemcacheLockBackend',
+        'arguments' => ['container', '@memcache.backend.cache.container'],
+      ],
+      'cache_tags_provider.container' => [
+        'class' => 'Drupal\Core\Cache\DatabaseCacheTagsChecksum',
+        'arguments' => ['@database'],
+      ],
+      'cache.container' => [
+        'class' => 'Drupal\memcache\MemcacheBackend',
+        'arguments' => [
+          'container',
+          '@memcache.backend.cache.container',
+          '@lock.container',
+          '@memcache.config',
+          '@cache_tags_provider.container',
         ],
       ],
-    ];
+    ],
+  ];
+
+  // Override default fastchained backend for static bins.
+  // @see https://www.drupal.org/node/2754947
+  $settings['cache']['bins']['bootstrap'] = 'cache.backend.memcache';
+  $settings['cache']['bins']['discovery'] = 'cache.backend.memcache';
+  $settings['cache']['bins']['config'] = 'cache.backend.memcache';
```

Comment:
@lcatlett Question, have you done any performance comparisons of using `memcache` vs the default `chainedfast` for `cache.backend` on these three bins? In my initial testing, using `memcache` is results in worse performance. 

I can always open a separate issue for this, but would be curious to hear if you have noticed the same.

Reply:
@greylabel if you are using PHP 7, it is very likely that performance seems degraded with memcache vs. chainedfast because of large objects in the discovery and config bins - since compression is disabled by default t

Reply:
I am very interested to try this with the compression settings added in this PR. I also was just looking at the `compressed_cache`, interesting as well. If I find time I will do some more formal benchmarking.